common/src/main/java/com/viaversion/viaversion/protocol/ProtocolManagerImpl.java

```diff
@@ -176,39 +171,32 @@ public void registerProtocol(Protocol protocol, ProtocolVersion clientVersion, P
 
     @Override
     public void registerProtocol(Protocol protocol, List<Integer> supportedClientVersion, int serverVersion) {
-        // Register the protocol's handlers
-        protocol.initialize();
+        registerProtocolSupplier(protocol.getClass(), () -> protocol, supportedClientVersion, serverVersion);
+        getProtocol(protocol.getClass());  // Api guarantees initialisation
```

Comment:
No need to call getProtocol here if there's no actual initialisation anyways - but to make its use clearer and avoid unnecessary synchronization, I'd suggest the following: Make CachedProtocolSupplier an interface and add a second implementation only holding a final protocol that is directly returned, which is then used here

Reply:
Did change it in an abstract class instead of an interface due to member specific function calls not included in the ProtocolManager-API in the initialisation code (I consider loading the mappings, protocol.initialize and protocol.register as some sort of initialization).